homeassistant/components/sensor/xiaomi_miio.py

```diff
@@ -0,0 +1,209 @@
+        """Map services to methods on XiaomiAirQualityMonitor."""
+        method = SERVICE_TO_METHOD.get(service.service)
+        params = {key: value for key, value in service.data.items()
+                  if key != ATTR_ENTITY_ID}
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+        if entity_ids:
+            devices = [device for device in hass.data[DATA_KEY].values() if
+                       device.entity_id in entity_ids]
+        else:
+            devices = hass.data[DATA_KEY].values()
+
+        update_tasks = []
+        for device in devices:
+            yield from getattr(device, method['method'])(**params)
+            update_tasks.append(device.async_update_ha_state(True))
+
+        if update_tasks:
+            yield from asyncio.wait(update_tasks, loop=hass.loop)
+
+    for service in SERVICE_TO_METHOD:
+        schema = SERVICE_TO_METHOD[service].get('schema', SERVICE_SCHEMA)
+        hass.services.async_register(
+            DOMAIN, service, async_service_handler, schema=schema)
+
+
+class XiaomiAirQualityMonitor(Entity):
+    """Representation of a Xiaomi Air Quality Monitor."""
+
+    def __init__(self, name, device, model):
+        """Initialize the entity."""
+        self._name = name
+        self._model = model
+        self._icon = 'mdi:cloud'
+        self._unit_of_measurement = 'AQI'
+
+        self._device = device
+        self._state = None
+        self._state_attrs = {
+            ATTR_POWER: None,
+            ATTR_BATTERY_LEVEL: None,
+            ATTR_CHARGING: None,
+            ATTR_TIME_STATE: None,
+            ATTR_MODEL: self._model,
+        }
+
+    @property
+    def should_poll(self):
+        """Poll the miio device."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of this entity, if any."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the icon to use for device if any."""
+        return self._icon
+
+    @property
+    def available(self):
+        """Return true when state is known."""
+        return self._state is not None
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        return self._state_attrs
+
+    @asyncio.coroutine
+    def _try_command(self, mask_error, func, *args, **kwargs):
+        """Call a device command handling error messages."""
+        from miio import DeviceException
+        try:
+            result = yield from self.hass.async_add_job(
+                partial(func, *args, **kwargs))
+
+            _LOGGER.debug("Response received from miio device: %s", result)
+
+            return result == SUCCESS
```

Comment:
Do you think we should respond with a boolean inside python-miio for this device instead of passing the result back?

Reply:
No. The return value is needed for the status call.

Reply:
I just meant that whether this should be modified to respond differently on python-miio's end. But it's fine as it is.